README.md

```diff
@@ -21,11 +21,96 @@ are not returned to the client process address space, and in the case
 of Secure Peripheral or TEE application backends, they are not returned
 to the system daemon process address space after initial storage).
 
-## Building requirements:
+## Components
+
+### Secrets daemon
+
+We have a system service (daemon) called `sailfishsecretsd` which listens to connections over
+peer-to-peer D-Bus and has two responsibilities:
+
+* Store (and retrieve) sensitive data in secure locations
+* Perform cryptographic operations on behalf of clients
+
+Rationale:
+
+* The user's secret keys are never actually exposed to the client application, so
+the user doesn't have to trust the client to handle sensitive data correctly. In fact, if
+you have a plugin that enables the use of a suitable peripheral, the key is never even loaded
+into system memory.
+* Clients don't have to "know about cryptography" or be certified, because all crypto
+operations are done by the daemon.
```

Comment:
are done by plugins loaded by the daemon

Reply:
Will fix.